cmd/dist-receiver/main.go

```diff
@@ -0,0 +1,271 @@
+	// 4: target directory
+
+	stageNow = 0
+	sig = make(chan bool)
+
+	server := &http.Server{
+        Addr:         "0.0.0.0:4002",
+        ReadTimeout:  5 * time.Second,
+        WriteTimeout: 5 * time.Second,
+        IdleTimeout:  600 * time.Second,
+    }
+
+	http.HandleFunc("/stage", stage)
+	http.HandleFunc("/next", next)
+	http.HandleFunc("/connect", connect)
+	go func() {
+		if err := server.ListenAndServe(); err != nil {
+			panic(fmt.Errorf("failed to spawn command receiver: %s", err))
+		}
+	}()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	node, _, err := spawnEphemeral(ctx, args[1])
+	if err != nil {
+		panic(fmt.Errorf("failed to spawn ephemeral node: %s", err))
+	}
+
+	if err := connectToPeers(ctx, node, []string{args[1]}); err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to connect to root peer: {}", err)
+	}
+
+	stageNow = 1
+
+	<-sig
+
+	if err := connectToPeers(ctx, node, targets); err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to connect to peers: {}", err)
+	}
+
+	cid := icorepath.New(args[2])
+
+	if err := node.Dht().Provide(ctx, cid); err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to seed the resource file: {}", err)
+	}
+
+	if err := node.Pin().Add(ctx, cid); err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to pin the resource file: {}", err)
+	}
+
+	rootNode, err := node.Unixfs().Get(ctx, cid)
+
+	if err != nil {
+		panic(fmt.Errorf("could not get file with CID: %s", err))
+	}
+
+	if err := os.RemoveAll(args[3]); err != nil {
+		panic(fmt.Errorf("could not clean previous temporary file: %s", err))
+	}
+
+	err = files.WriteTo(rootNode, args[3])
+	if err != nil {
+		panic(fmt.Errorf("could not write out the fetched CID: %s", err))
+	}
+
+	err = os.MkdirAll(args[4], os.ModePerm)
+	if err != nil {
+		panic(fmt.Errorf("failed to create target directory: %s", err))
+	}
+
+	// the following command needs variable aurgments anyway, which will cause G204 in gosec
+	cmd := exec.Command("tar", "-C", args[4], "-xzf", args[3]) // #nosec G204
+	_, err = cmd.Output()
+	if err != nil {
+		panic(fmt.Errorf("failed to uncompress resource file: %s", err))
+	}
+
+	stageNow = 2
+
+	<-sig
+
+	// the following command needs variable aurgments anyway, which will cause G204 in gosec
+	exec.Command("rm", args[3]) // #nosec G204
+}
+
+func createNode(ctx context.Context, repoPath string) (*core.IpfsNode, error) {
+	// Open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the node
+	nodeOptions := &core.BuildCfg{
+		Online:  true,
+		Routing: libp2p.DHTOption, // This option sets the node to be a full DHT node (both fetching and storing DHT Records)
+		// Routing: libp2p.DHTClientOption, // This option sets the node to be a client DHT node (only fetching records)
+		Repo: repo,
+	}
+
+	return core.NewNode(ctx, nodeOptions)
+}
+
+func spawnEphemeral(ctx context.Context, bootstrap string) (icore.CoreAPI, *core.IpfsNode, error) {
+	err := setupPlugins("")
```

Comment:
if err := setupPlugins(""); err != nil {
		return nil, nil, err
	}
	
	This way of writing gives you more control over the scope of the variable

Reply:
Thanks, updated